tf2_ros/include/tf2_ros/message_filter.h

```diff
@@ -353,8 +369,8 @@ class MessageFilter : public MessageFilterBase, public message_filters::SimpleFi
             info.handles.push_back(next_handle_index_++);
           }
         }
-        catch (...) {
-          // never transformable
+        catch (std::exception & e)  {
```

Comment:
Could we catch a better exception than this?

Reply:
Unfortunately, looking at this closer, I'm not sure we can.  The documentation for `std::shared_future.get()` (which is the type of `future`) says [this](https://en.cppreference.com/w/cpp/thread/shared_future/get):


```
If an exception was stored in the shared state referenced by the future (e.g. via a call to std::promise::set_exception()) then that exception will be thrown. 
```

Since we don't know what `waitForTransform` will do with the future, we have to assume it can set anything in there, and thus we must just catch `std::exception`, as far as I can tell.

That being said, this can be a const reference, I believe.

Reply:
I have added const to the ref, I figure it's at least a little better than `(...)` so we're making incremental progress.